File: ibkr_show_backend/app/schemas/trade_decision.py
from pydantic import BaseModel, Field


class TradeDecisionHealthResponse(BaseModel):
    enabled: bool
    llm_configured: bool
    longbridge_configured: bool
    mcp_enabled: bool = False
    mcp_available: bool = False
    mcp_auth_status: str = "disabled"
    mcp_last_error: str = ""
    sdk_fallback_available: bool = False
    longbridge_sdk_configured: bool = False
    public_data_mode: str = "unavailable"
    trade_review_available: bool
    account_data_source: str
    public_market_data_source: str
    agent_mode: str = "trade_decision_langgraph_v1"
    graph_version: str = "trade_decision_graph_v1"
    message: str


class TradeDecisionHoldingItem(BaseModel):
    symbol: str
    normalized_symbol: str
    quantity: float | None = None
    avg_cost: float | None = None
    current_price: float | None = None
    market_value: float | None = None
    position_pct: float | None = None
    unrealized_pnl: float | None = None
    unrealized_pnl_pct: float | None = None
    latest_review_score: float | None = None
    latest_decision: str | None = None
    data_source: str = "IBKR"


class TradeDecisionHoldingsResponse(BaseModel):
    items: list[TradeDecisionHoldingItem]


class TradeDecisionAnalyzeHoldingRequest(BaseModel):
    question: str | None = None
    force_refresh: bool = False


class TradeDecisionAnalyzeEntryRequest(BaseModel):
    symbol: str
    question: str | None = None
    force_refresh: bool = False


class TradeDecisionScoreItem(BaseModel):
    score: float
    max_score: float
    reason: str = ""


class TradeDecisionPositionAdvice(BaseModel):
    current_position_pct: float | None = None
    suggested_target_position_pct: float | None = None
    max_position_pct: float | None = None
    suggested_cash_amount: float | None = None
    position_size_label: str


class TradeDecisionExecutionStep(BaseModel):
    step: int | None = None
    condition: str | None = None
    action: str | None = None
    amount: float | None = None
    note: str | None = None


class TradeDecisionExecutionPlan(BaseModel):
    should_act_now: bool
    plan: list[dict] = Field(default_factory=list)
    invalid_conditions: list[str] = Field(default_factory=list)
    recheck_triggers: list[str] = Field(default_factory=list)


class AgentRunTraceItem(BaseModel):
    event: str
    node_name: str | None = None
    tool: str | None = None
    tool_call_id: str | None = None
    round: int | None = None
    arguments: dict | None = None
    steps: list[str] | None = None
    ok: bool | None = None
    summary: str | None = None
    latency_ms: int | None = None
    created_at_ms: int | None = None
    elapsed_ms: int | None = None
    tools_called: list[str] | None = None
    tool_call_count: int | None = None
    tool_calls: list[dict] | None = None
    rounds_used: int | None = None
    fallback_used: bool | None = None
    fallback_reason: str | None = None
    structured_output: dict | None = None


class TradeDecisionResult(BaseModel):
    id: str
    decision_type: str
    symbol: str
    user_question: str | None = None
    overall_score: float
    rating: str
    action: str
    confidence: str
    decision_summary: str
    score_detail: dict[str, TradeDecisionScoreItem]
    position_advice: TradeDecisionPositionAdvice
    execution_plan: TradeDecisionExecutionPlan
    key_reasons: list[str]
    major_risks: list[str]
    review_warnings: list[str]
    data_limitations: list[str]
    evidence_used: list[str]
    data_source_summary: dict
    card_pack: dict = Field(default_factory=dict)
    run_trace: list[AgentRunTraceItem] = Field(default_factory=list)
    metadata: dict = Field(default_factory=dict)
    evidence_summary: dict = Field(default_factory=dict)
    run_trace_summary: dict = Field(default_factory=dict)
    fallback_used: bool = False
    fallback_reason: str | None = None
    llm_error_summary: dict = Field(default_factory=dict)
    created_at: str
    updated_at: str


class TradeDecisionListResponse(BaseModel):
    items: list[TradeDecisionResult]

File: ibkr_show_frontend/src/types/tradeDecision.ts
export interface TradeDecisionHealth {
  enabled: boolean
  llm_configured: boolean
  longbridge_configured: boolean
  mcp_enabled: boolean
  mcp_available: boolean
  mcp_auth_status: string
  mcp_last_error: string
  sdk_fallback_available: boolean
  longbridge_sdk_configured: boolean
  public_data_mode: string
  trade_review_available: boolean
  account_data_source: string
  public_market_data_source: string
  message: string
}

export interface TradeDecisionScoreItem {
  score: number
  max_score: number
  reason: string
}

export interface TradeDecisionPositionAdvice {
  current_position_pct: number | null
  suggested_target_position_pct: number | null
  max_position_pct: number | null
  suggested_cash_amount: number | null
  position_size_label: string
}

export interface TradeDecisionExecutionPlan {
  should_act_now: boolean
  plan: Array<Record<string, unknown>>
  invalid_conditions: string[]
  recheck_triggers: string[]
}

export interface AgentRunTraceItem {
  event: string
  tool?: string | null
  tool_call_id?: string | null
  round?: number | null
  arguments?: Record<string, unknown> | null
  steps?: string[] | null
  ok?: boolean | null
  summary?: string | null
  latency_ms?: number | null
  created_at_ms?: number | null
}

export interface TradeDecisionResult {
  id: string
  decision_type: string
  symbol: string
  user_question: string | null
  overall_score: number
  rating: string
  action: string
  confidence: string
  decision_summary: string
  score_detail: Record<string, TradeDecisionScoreItem>
  position_advice: TradeDecisionPositionAdvice
  execution_plan: TradeDecisionExecutionPlan
  key_reasons: string[]
  major_risks: string[]
  review_warnings: string[]
  data_limitations: string[]
  evidence_used: string[]
  data_source_summary: Record<string, string>
  run_trace: AgentRunTraceItem[]
  metadata: Record<string, unknown>
  evidence_summary: Record<string, unknown>
  run_trace_summary: Record<string, unknown>
  fallback_used?: boolean
  fallback_reason?: string | null
  created_at: string
  updated_at: string
}

export interface TradeDecisionListResponse {
  items: TradeDecisionResult[]
}

export interface TradeDecisionHoldingItem {
  symbol: string
  normalized_symbol: string
  quantity: number | null
  avg_cost: number | null
  current_price: number | null
  market_value: number | null
  position_pct: number | null
  unrealized_pnl: number | null
  unrealized_pnl_pct: number | null
  latest_review_score: number | null
  latest_decision: string | null
  data_source: string
}

export interface TradeDecisionHoldingsResponse {
  items: TradeDecisionHoldingItem[]
}
